src/lib/dom-api.html

```diff
@@ -593,11 +593,73 @@
 
     };
 
+    Object.defineProperties(DomApi.prototype, {
+      activeElement: {
+        configurable: true,
+        get: function() {
+          var ownerDocument = this.node.ownerDocument;
+
+          if (ownerDocument || !window.ShadowDOMPolyfill) {
+            return this.node.activeElement;
+          } else {
+            return window.ShadowDOMPolyfill.wrapIfNeeded(this.node).activeElement;
+          }
+        }
+      }
+    });
+
     // changes and accessors...
     if (!Settings.useShadow) {
 
       Object.defineProperties(DomApi.prototype, {
 
+        activeElement: {
+          get: function() {
+            var active = document.activeElement;
+            if (!active) return null;
+
+            var nodeIsDocument = this.node === document;
+
+            if (!nodeIsDocument) {
+              // If this node isn't a document or shady root, then it doesn't
+              // have an active element.
+              if (!this.node._isShadyRoot) return null;
+
+              // If this shady root's host is the active element or the active
+              // element is not a descendant of the host, then it doesn't have
+              // an active element.
+              if (this.node.host === active || !this.node.host.contains(active)) {
+                return null;
+              }
+            }
+
+            // This node is either the document or a shady root of which the
+            // active element is a descendant of its host; iterate upwards to
+            // find the active element's most shallow host.
+            var activeRoot = Polymer.dom(active).getOwnerRoot();
+            while (activeRoot && activeRoot !== this.node) {
+              active = activeRoot.host;
+              activeRoot = Polymer.dom(active).getOwnerRoot();
+            }
+
+            // This node is the document and we've found the active element's
+            // most shallow host.
+            if (nodeIsDocument && !activeRoot) {
+              return active;
+            }
+
+            // This node is a shady root and we've found the active element's
+            // most shallow host (within the shady root).
+            if (!nodeIsDocument && activeRoot === this.node) {
+              return active;
+            }
+
+            // The active element is within the light DOM of this shady root.
+            return null;
```

Comment:
As you noted, the expectation here is to return active. So it seems like the 2 conditionals above are not necessary and we can always just return `active` if we get to here.